drivers/hmis/app/graphql/types/base_audit_event.rb

```diff
@@ -0,0 +1,57 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Types
+  class BaseAuditEvent < BaseObject
+    def self.create(node_class)
```

Comment:
Should this maybe be `new` or `initialize` instead of create since it's not saving anything? (or maybe it is?)

Reply:
This is a GraphQL class, not an AR model, so the create method here is doing totally different stuff than in activerecord. In this case this method is basically working as a factory. I pass it some parameters and then it uses those parameters to construct a new class that inherits from this class, which represents a new GraphQL type. It's not doing anything with DB persistence here.

Reply:
Yeah, that was my point.  It seemed odd to use `create` instead of `new` when there _is_ no persistence.  I'm happy to leave it, but to me `create` comes with some connotation of persistence.

Reply:
There's no DB persistence, but it does create a new GraphQL type in the runtime. While it's not persistent in terms of actually saving something in the DB, the type that's created does act as a persistent part of the GraphQL schema, so I can see an argument for using `create`, but I could go either way. We are using this terminology in other places in the schema, so this is currently consistent with that, but we could definitely change it if you all would prefer. Let me know what you want to do.

Reply:
Gotcha, that makes sense. `new` or `initialize` seems better. lets make it consistent across all similar classes though - `BasePaginated`, `BaseAccess`, `BaseAuditEvent`

Reply:
I don't think either `new` or `initialize` are appropriate here. This method doesn't instantiate a new class instance like those terms imply. This method creates and returns an entirely new subclass, which it also does not instantiate. I'd prefer something like `build` that more clearly indicates its nature as a class factory

Reply:
I'll go ahead and make that change here